linkedin/phantom_api/phantombuster_api.py

```
@@ -0,0 +1,236 @@
+        ```
+                      id            name scriptId ... lastEndStatus queuedContainers runningContainers
+        2862499141527492   Search Export     6988           success                0                 0
+        3593602419926765 Profile Scraper     3112           success                0                 0
+        3933308360008191 Profile Scraper     3112           success                0                 0
+        # pylint: enable=line-too-long
+        ```
+        """
+        data = self._get_phantom_data()
+
+        if data and "data" in data and "agents" in data["data"]:
+            return pd.DataFrame(data["data"]["agents"])
+        #
+        _LOG.error("The data structure is invalid: Phantoms not found.")
+        return None
+
+    def download_result_csv_by_phantom_id(
+        self, phantom_id: str, output_path: str
+    ) -> None:
+        """
+        Download result CSV by phantom ID.
+
+        :param phantom_id: The ID of the phantom for which to fetch the result CSV
+        :param output_path: The path to save the result CSV
+        """
+        result_url = self._get_result_csv_by_phantom_id(phantom_id)
+        if result_url:
+            self._download_result_csv_to_local(result_url, output_path)
+
+    # #########################################################################
+
+    @staticmethod
+    def _download_result_csv_to_local(
+        result_csv_url: str, output_path: str
+    ) -> None:
+        """
+        Download result CSV from Phantombuster server to local storage.
+
+        :param result_csv_url: The URL of the result CSV
+        :param output_path: The path to save the result CSV
+        """
+        hio.create_enclosing_dir(output_path, incremental=True)
+        try:
+            response = requests.get(result_csv_url, timeout=30)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            _LOG.error("%s", e)
+            return None
```

You are using a lot of `None` which complicates the control flow and pollutes the interfaces...
This was done a lot in C (e.g., if you look at Linux kernel)

IMO it's better to raise an exception.